domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -434,5 +585,40 @@ class ExplorationProgressController @Inject constructor(
     // Advance the stage, but do not notify observers since the current state can be reported
     // immediately to the UI.
     progress.advancePlayStageTo(ExplorationProgress.PlayStage.VIEWING_STATE)
+
+    // Mark a checkpoint in the exploration once the exploration has loaded.
+    saveExplorationCheckpoint()
+  }
+
+  private fun markExplorationAsInProgressSaved(
+    profileId: ProfileId,
+    topicId: String,
+    storyId: String,
+    explorationId: String,
+    lastPlayedTimestamp: Long
+  ) {
+    storyProgressController.recordChapterAsInProgressSaved(
+      profileId,
+      topicId,
+      storyId,
+      explorationId,
+      lastPlayedTimestamp
+    )
+  }
+
+  private fun markExplorationAsInProgressNotSaved(
+    profileId: ProfileId,
+    topicId: String,
+    storyId: String,
+    explorationId: String,
+    lastPlayedTimestamp: Long
+  ) {
+    storyProgressController.recordChapterAsInProgressNotSaved(
```

Comment:
This seems like a dependency inversion, but I'm not sure what the alternative is at the moment.